### PR TITLE
refactor: moved LocationServices to services

### PR DESCRIPTION
**Pull request by LinMoskovitch**
Also moved all location services part to this class instead of methods in WelcomePage

### CODE REVIEW

File: Notify/Notify/Notify/ViewModels/WelcomePageViewModel.cs [hunk 1]
```diff
@@ -53,7 +55,7 @@ private async void onLoginClicked()
             bool debugAutoLogin = true;
             IsBusy = true;
             
-            manageLocationTracking();
+            locationService.ManageLocationTracking();
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/ViewModels/WelcomePageViewModel.cs:58]: I think this should only be called after the user has logged in, and not before we're checking if he's authenticated.

Comment (LinMoskovitch) [reply Notify/Notify/Notify/ViewModels/WelcomePageViewModel.cs:58]: Right

File: Notify/Notify/Notify/Services/Location/LocationServices.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        {
+            if (Device.RuntimePlatform == Device.Android)
+            {
+                MessagingCenter.Subscribe<LocationMessage>(this, "Location",
+                    message =>
+                    {
+                        Device.BeginInvokeOnMainThread(() =>
+                        {
+                            Console.WriteLine(
+                                $"{message.Latitude}, {message.Longitude}, {DateTime.Now.ToLongTimeString()}");
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/Services/Location/LocationServices.cs:114]: I think we should add "User's current location:" in the beginning of the message, and to print it to console only when we're on debug mode.

File: Notify/Notify/Notify/Services/Location/LocationServices.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                            MessagingCenter.Send(errormessage, "LocationError");
+                        });
+                    }
+                }
+            }, token);
+        }
+
+        public void Current_PositionChanged(object sender, Plugin.Geolocator.Abstractions.PositionEventArgs e)
+        {
+            Console.WriteLine($"{e.Position.Latitude}, {e.Position.Longitude}, {e.Position.Timestamp.TimeOfDay}");
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/Services/Location/LocationServices.cs:54]: Same as I suggested before.

Comment (LinMoskovitch) [reply Notify/Notify/Notify/Services/Location/LocationServices.cs]: Done

File: Notify/Notify/Notify/Services/Location/LocationServices.cs (hunk 2 above)
Comment (LinMoskovitch) [reply Notify/Notify/Notify/Services/Location/LocationServices.cs]: Done

**Review by DekelR — APPROVED** [review]
looks ok - but didn't try to download locally and run

File: Notify/Notify/Notify/Services/Location/LocationServices.cs [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                }
+            }
+        }
+        
+        public void StartService()
+        {
+            var startServiceMessage = new StartServiceMessage();
+            MessagingCenter.Send(startServiceMessage, "ServiceStarted");
+            Preferences.Set("LocationServiceRunning", true);
+            Console.WriteLine("Location Service has been started!");
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/Services/Location/LocationServices.cs:143]: I think we should use Debug.WriteLine() here as well.

**Pull request merged**